### PR TITLE
nautilus: qa/tasks/cephfs: py 3 transition fixes 

### PR DESCRIPTION
**Pull request by ajarr**
Fixes: https://tracker.ceph.com/issues/45966
Fixes: https://tracker.ceph.com/issues/45958

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephfs/test_exports.py [hunk 1]
```diff
@@ -131,12 +131,12 @@ def test_export_pin_getfattr(self):
             p = self.mount_a.client_remote.sh('uname -r', wait=True)
             dir_pin = self.mount_a.getfattr("1", "ceph.dir.pin")
             log.debug("mount.getfattr('1','ceph.dir.pin'): %s " % dir_pin)
-            if str(p.stdout.getvalue()) < "5" and not(dir_pin):
+            if str(p) < "5" and not(dir_pin):
                 self.skipTest("Kernel does not support getting the extended attribute ceph.dir.pin")
-        self.assertTrue(self.mount_a.getfattr("1", "ceph.dir.pin") == "1")
-        self.assertTrue(self.mount_a.getfattr("1/2", "ceph.dir.pin") == "0")
+        self.assertEqual(self.mount_a.getfattr("1", "ceph.dir.pin"), b'1')
```
[the remark this answers is not in the document]
Comment (ajarr) [reply qa/tasks/cephfs/test_exports.py:136]: Right. Hit this, 
http://pulpito.ceph.com/yuriw-2020-06-10_18:00:48-multimds-wip-yuri3-testing-2020-06-10-1530-nautilus-distro-basic-smithi/5134200/
```
AssertionError: '1' != b'1'
```

**Comment by ajarr** [comment]
@tchaikov, the kcephfs suite job failures are fixed with the commit https://github.com/ceph/ceph/pull/35520/commits/ab1277227218bb01b61b2db0d68eb961a37610fd that is in this PR.
Would you be OK to approve this PR if I removed the commit that does not fix the test_exports failure? I can then ask Yuri to merge this PR

kcephfs suite jobs,
http://pulpito.ceph.com/yuriw-2020-06-10_17:59:34-kcephfs-wip-yuri3-testing-2020-06-10-1530-nautilus-distro-basic-smithi/

**Comment by ajarr** [comment]
I've updated the PR with only the kcephfs suite job failures fix.

File: qa/tasks/cephfs/kernel_mount.py [hunk 2]
```diff
@@ -210,10 +210,10 @@ def get_id_to_dir():
             print(json.dumps(get_id_to_dir()))
             """)
 
-        p = self.client_remote.run(args=[
-            'sudo', 'python', '-c', pyscript
-        ], stdout=StringIO(), timeout=(5*60))
-        client_id_to_dir = json.loads(p.stdout.getvalue())
+        output = self.client_remote.sh([
```
Comment (tchaikov) [inline qa/tasks/cephfs/kernel_mount.py:213]: this change should have two backport of follow commits

- 3c751fea7f48ef613286211797a7eafc4913b36e
- 875b45f3c903781ba99dbd2fcf4d2fc89001c25b

the root cause is

- i forgot to backport 3c751fea7f48ef613286211797a7eafc4913b36e
- i failed to correctly backport 875b45f3c903781ba99dbd2fcf4d2fc89001c25b to nautilus correctly

Comment (ajarr) [reply qa/tasks/cephfs/kernel_mount.py:213]: Thanks a lot, Kefu!

**Comment by ajarr** [comment]
@tchaikov PTAL

File: qa/tasks/cephfs/kernel_mount.py (hunk 2 above)
Comment (tchaikov) [reply qa/tasks/cephfs/kernel_mount.py:213]: please either backport 875b45f3c903781ba99dbd2fcf4d2fc89001c25b or note down the reason why this change is not cherry-picked from master in the commit message.

Comment (ajarr) [reply qa/tasks/cephfs/kernel_mount.py:213]: Mentioned in the commit message that it was an incomplete backport and that it missed some commit line.

**Comment by ajarr** [comment]
Backporting commit 3c751fea7f48ef613286211797a7eafc4913b36e based on @tchaikov suggestion introduced/exposed new set of job failures in kcephfs and fs suite of Yuri's another round of testing,
http://pulpito.ceph.com/?branch=wip-yuri3-testing-2020-06-11-1608-nautilus

I was able to track down additional missing backport commits. They are added in this PR,
https://github.com/ceph/ceph/pull/35520/commits/1f302b7064c9e289d40910f745997844c5ed7d95
https://github.com/ceph/ceph/pull/35520/commits/2d15a1024f6cf2b58bc5719cec2996e9cc263c05

**Comment by ajarr** [comment]
@tchaikov PTAL

**Pull request merged**